### DISCUSSION TITLE
How to have separate validation for 2 input fields in child component with Controller

### DISCUSSION THREAD

**Discussion by PritamSangani** [react-hook-form/react-hook-form#2613] [Q&A]
Hi. 
I am using Material UI for my form inputs. 
I have a parent form component and child components for each section of the form. One particular section has 2 input fields, both are required fields. I am wondering whether it is possible to use Controller in this use case to render the child component for this section, or whether the best way to achieve separate validation (+ messages) is using refs which I would pass down to the child component - one for each input.

Thanks.

**Comment by bluebill1049** [comment]
please provide a simple CSB.

**Reply by PritamSangani** [reply]
Here's a simple CSB showing my use case. I have added a couple of comments in the App.tsx file to explain the questions I have.

**Reply by bluebill1049** [reply]
i think you forgot to post the link.

**Reply by PritamSangani** [reply]
Doh! 🤦 Here it is: https://codesandbox.io/s/great-visvesvaraya-6os7k

**Reply by PritamSangani** [reply]
Hi @bluebill1049. Have you had a chance to look at this yet?

**Reply by bluebill1049** [reply]
errors will only occur one validation rule at a time unless change `criteriaMode`, and your rules looks fine.

**Reply by PritamSangani** [reply]
Thanks. I think that worked. I am however getting a TypeScript error saying that types property doesn't exist on the DeepMap. And also I am wondering whether it is best practice to use a single Controller for 2 input text fields or register both input fields separately thereby keeping the errors object separate?

**Reply by PritamSangani** [reply]
Hi @bluebill1049. Ever since I upgraded the react-hook-form library to later than 6.4.0, I am getting type errors, when I use NestedValue and when I try and get `types` property from the `errors.{fieldName}` object. Can you take a look and see what I am doing wrong, as this did work in version 6.4.0. https://codesandbox.io/s/great-visvesvaraya-6os7k?file=/src/App.tsx.

**Reply by bluebill1049** [reply]
please provide the steps to reproduce.
